include/pagmo/algorithms/gwo.hpp

```diff
@@ -0,0 +1,351 @@
+
+                if(newfitness > alpha_score && newfitness > beta_score && newfitness <  delta_score){
+                    delta_score = newfitness;
+                    delta_pos = agents_position[i];
+
+                }
+               
+        
+            }
+             a= 2-gen*((2)/m_gen);
+
+           
+            for (decltype(m_agent) i = 0u; i < m_agent; ++i){
+                  // Encircling prey and attack
+              for(decltype(dim) k = 0; k < dim; ++k){
+                    r1 = drng(m_seed);
+                    r2 = drng(m_seed);
+
+                    a_vector[0] = 2*a*r1-a;  // Equation (3.3)
+                    c_vector[0] = 2*r2; // Equation (3.4)
+
+                    d_vector[0] = std::cmath::abs(c_vector[0] * alpha_pos[k]- agents_position[i][k]); // Equation (3.5)-part 1
+                    x_vector[0] = alpha_pos[k]- a_vector[0] * d_vector[0]; // Equation (3.6)-part 1
+
+                    r1 = drng(m_seed);
+                    r2 = drng(m_seed);
+
+                    a_vector[1] =2*a*r1-a; 
+                    c_vector[1]=2*r2;
+
+                    d_vector[1]=abs(c_vector[1]*beta_pos[k]-agents_position[i][k]); // Equation (3.5)-part 2
+                    x_vector[1]= beta_pos[k]-a_vector[1] * d_vector[1]; // Equation (3.6)-part 2       
+                
+                    r1=random.random()
+                    r2=random.random() 
+                
+                
+                    a_vector[2] =2*a*r1-a; 
+                    c_vector[2]=2*r2;
+
+                    d_vector[2]=abs(c_vector[2]*delta_pos[k]-agents_position[i][k]); // Equation (3.5)-part 3
+                    x_vector[2]= delta_pos[k]-a_vector[2] * d_vector[2]; // Equation (3.6)-part 3          
+                
+                agents_position[i][k] = (x_vector[0]+x_vector[1]+x_vector[2])/3;  // Equation (3.7)
+
+
+              } 
+            }  // End of one iteration
+/// Single entry of the log (gen, alpha, beta, delta)
+              // Logs and prints (verbosity modes > 1: a line is added every m_verbosity generations)
+            if (m_verbosity > 0u) {
+                // Every m_verbosity generations print a log line
+                if (gen % m_verbosity == 1u || m_verbosity == 1u) {
+                    // Every 50 lines print the column names
+                    if (count % 50u == 1u) {
+                        print("\n", std::setw(7), "Gen:", std::setw(15), "Alpha:", std::setw(15),
+                              "Beta:", std::setw(15), "Delta:", '\n');
+                    }
+                    print(std::setw(7), gen, std::setw(15), alpha_score, std::setw(15),
+                          beta_score, std::setw(15), delta_score, '\n');
+                    ++count;
+                    // Logs
+                    m_log.emplace_back(gen, alpha_score, beta_score, delta_score);
+                }
+            }
+
+        }// end main gmo iterations
+         
+        
+
+        return pop;
```

Comment:
indeed ... but in your code you have not update it :)

Reply:
I try to compile the quick example with gwo and found it come up with type error that it require return type population, is it a convention of pagmo2?

Reply:
Yes the fitness method must return the evolved pop ... that's why I mentioned you must call the pop.set_xf method to insert an evolved individual into the pop